adoc/chapters/programming_interface.adoc

```diff
@@ -17640,40 +17635,511 @@ member function template.
+!====
+a!
+[source]
+----
+#ifdef __SYCL_DEVICE_ONLY__
+operator vector_t() const
+#endif
+
+operator DataT() const
+static constexpr size_t byte_size() noexcept
+static constexpr size_t size() noexcept
+
+size_t get_size() const   // Deprecated
+size_t get_count() const  // Deprecated
+
+template <typename ConvertT,
+          rounding_mode RoundingMode = rounding_mode::automatic>
+vec<ConvertT, NumElements> convert() const
+
+template <typename asT> asT as() const
+
+template <access::address_space AddressSpace, access::decorated IsDecorated>
+void store(size_t offset, multi_ptr<DataT, AddressSpace, IsDecorated> ptr) const
+----
+!====
+_Constraints:_ These functions have the same constraints as the equivalent
+member functions of the [code]#vec# class.
+
+_Effects:_ The effect of these functions is the same as if they were applied to
+a temporary [code]#vec# object that contains the result of the swizzle
+operation.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+operator vec<DataT, NumElements>() const
+----
+!====
+_Constraints:_ Available only when [code]#NumElements > 1#.
+
+_Returns:_ A new [code]#vec# object that represents the elements of the
+[code]#vec# object that underlies the [code]#+__swizzled_vec__+# view with the
+swizzle operation applied.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+template <int... swizzleIndexes> __swizzled_vec__</*unspecified*/> swizzle() const
+
+__swizzled_vec__</*unspecified*/> XYZW_ACCESS() const
+__swizzled_vec__</*unspecified*/> RGBA_ACCESS() const
+__swizzled_vec__</*unspecified*/> INDEX_ACCESS() const
+
+#ifdef SYCL_SIMPLE_SWIZZLES
+__swizzled_vec__</*unspecified*/> XYZW_SWIZZLE() const
+__swizzled_vec__</*unspecified*/> RGBA_SWIZZLE() const
+#endif
+
+__swizzled_vec__</*unspecified*/> lo() const
+__swizzled_vec__</*unspecified*/> hi() const
+__swizzled_vec__</*unspecified*/> odd() const
+__swizzled_vec__</*unspecified*/> even() const
+----
+!====
+_Constraints:_ These functions have the same constraints as the equivalent
+member functions of the [code]#vec# class.
+
+_Returns:_ A new [code]#+__swizzled_vec__+# view of the underlying [code]#vec#
+object, where the view represents the combination of two swizzle operations.
+The first is the swizzle operation represented by the [code]#+__swizzled_vec__+#
+view.
+The second is the swizzle operation defined by the member function.
+The indices used by the second swizzle are the indices produced by the first
+swizzle.
+For example, if the second swizzle references the first element, this means the
+element of the underlying [code]#vec# that corresponds to the first element
+produced by the first swizzle.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+template <access::address_space AddressSpace, access::decorated IsDecorated>
+void load(size_t offset, multi_ptr<const DataT, AddressSpace, IsDecorated> ptr)
+----
+!====
+_Effects:_ Loads values from memory into elements of the underlying [code]#vec#
+object.
```

Comment:
I agree this is shorter, but I think it's less clear.  I fear that if we just say "indexes are translated" then readers won't know what that translation means exactly.  I think the longer wording I have states this more clearly.

I have the same response for the next three suggestions below also.  They all shorten the wording by saying that "indexes are translated", but I think this may not be clear to all readers.

Reply:
I'd like to keep this the way it is.  I think it would be less clear if we describe the translation in general terms.  As it is now, we can tailor the description to each operation, which I think is clearer.